model/task/task.go

```diff
@@ -3869,18 +3894,15 @@ func HasMatchingTasks(versionID string, opts HasMatchingTasksOptions) (bool, err
 
 func getTasksByVersionPipeline(versionID string, opts GetTasksByVersionOptions) []bson.M {
 	var match bson.M = bson.M{}
-
-	// Allow searching by either variant name or variant display
-	if len(opts.Variants) > 0 {
-		variantsAsRegex := strings.Join(opts.Variants, "|")
-
-		match = bson.M{
-			"$or": []bson.M{
-				{BuildVariantDisplayNameKey: bson.M{"$regex": variantsAsRegex, "$options": "i"}},
-				{BuildVariantKey: bson.M{"$regex": variantsAsRegex, "$options": "i"}},
-			},
-		}
+	if !opts.IncludeBuildVariantDisplayName && opts.UseLegacyAddBuildVariantDisplayName {
+		grip.Alert(message.Fields{
```

Comment:
Do we expect this to get hit often in the case that the programmer error exists? This will log to ops-alerts and spam that requires a revert can be excessive. If we do want to log to ops-alerts it may be preferable to make this a regular error and create a splunk alert on it so we can throttle / turn it off if needed, or else just code read carefully to ensure we aren't in this position / maybe have an opts validation stage.

Reply:
This doesn't seem to be really consistent through out the codebase. https://github.com/evergreen-ci/evergreen/blob/2a2b5f433c464af9361753b5624bdcb8f572a779/model/patch/patch.go#L1068 
https://github.com/evergreen-ci/evergreen/blob/2a2b5f433c464af9361753b5624bdcb8f572a779/trigger/registry.go#L39
https://github.com/evergreen-ci/evergreen/blob/2a2b5f433c464af9361753b5624bdcb8f572a779/units/event_notifier.go#L250
https://github.com/evergreen-ci/evergreen/blob/2a2b5f433c464af9361753b5624bdcb8f572a779/units/host_monitoring_idle_termination.go#L196
 . Since either of these values can be dynamic and we use this function in so many places it will be difficult to notice this in code reviews.

Reply:
Done replaced this with an error and some validation.